tekstowo.py

```diff
@@ -0,0 +1,464 @@
+    
+    except requests.exceptions.RequestException as e:
+        print(f"An error occured: {e}")
+        return max_page
+
+    if req.ok:
+        soup = BeautifulSoup(req.content, 'lxml')
+
+        if soup:
+            for page in soup.find_all(class_='page-link'):
+                if page.text.isnumeric() and int(page.text) > max_page:
+                    max_page = int(page.text)
+
+    return max_page
+
+def create_lut_pagination() -> dict:
+    """
+    Creates a look-up table for each letter containing max page number
+    per letter in 'alphabet' variable. By default these are all26 letters 
+    of the alhpabet with an addition of 'pozostale' entry, which is a 
+    domain specific requirement.
+    """
+
+    alphabet = list(string.ascii_uppercase) + ['pozostale']
+
+    lut_pages = {}
+
+    for letter in alphabet:
+        time.sleep(1)
+        url = f"https://www.tekstowo.pl/artysci_na,{letter}.html"
+
+        try:
+            lut_pages[letter] = get_max_page_number(url)
+            print(f"Letter {letter} has {lut_pages[letter]} pages of artists.")
+
+        except Exception as e:
+            print(f"An error has occured for letter {letter}: {e}")
+   
+    return lut_pages
+
+def pages_per_letter(ARTIST_LETTER: str) -> dict:
+    """
+    Returns the value of the last page containing the songs
+    per artist.
+    """
+
+    letter_max_page = 0
+
+    if len(ARTIST_LETTER) < 2:
+        ARTIST_LETTER = ARTIST_LETTER.upper()
+    
+    url = f"https://www.tekstowo.pl/artysci_na,{ARTIST_LETTER}.html"
+
+    try:
+        letter_max_page = get_max_page_number(url)
+        print(f"Letter {ARTIST_LETTER} has {letter_max_page} pages of artists.")
+
+    except Exception as e:
+        print(f"An error has occured for letter {ARTIST_LETTER}: {e}")
+   
+    return letter_max_page
+
+def get_artists(ARTIST_LETTER: str, max_page_per_letter: Dict[str, int] or int) -> Tuple[List[str], int]:
+    """
+    Scrape all of the artists starting with a given letter in the alphabet.
+    """
+    
+    urls = []
+
+    if isinstance(max_page_per_letter, dict):
+        limit = max_page_per_letter[ARTIST_LETTER]
+    elif isinstance(max_page_per_letter, int):
+        limit = max_page_per_letter
+    else:
+        raise ValueError("max_page_per_letter should be either a dictionary \
+            with capital letters as keys and integers as values or a plain \
+            integer value.")
+
+    for page in range(1, limit + 1):
+        url = f"https://www.tekstowo.pl/artysci_na,{ARTIST_LETTER},strona,{page}.html"
+
+        try:
+            time.sleep(2)
+            response = requests.get(url, timeout=60)
+            if response.ok:
+                soup = BeautifulSoup(response.content, 'lxml')
+                for link in soup.find_all('a'):
+                    item = link.get('href')
+                    if isinstance(item, str) and 'piosenki_' in item:
+                        urls.append('https://tekstowo.pl' + item)
+
```

Comment:
A single space isn't need in the try-except block of code as it is one, continuous 'block entity'.